hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRevokeKycFromAccountHandler.java

```diff
@@ -40,27 +46,57 @@ public TokenRevokeKycFromAccountHandler() {
         // Exists for injection
     }
 
+    /**
+     * This method is called during the pre-handle workflow.
+     *
+     * @param context the {@link PreHandleContext} which collects all information
+     * @throws PreCheckException    for invalid tokens or if the token has no KYC key
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
         final var op = context.body().tokenRevokeKycOrThrow();
         final var tokenStore = context.createStore(ReadableTokenStore.class);
         final var tokenMeta = tokenStore.getTokenMeta(op.tokenOrElse(TokenID.DEFAULT));
-        if (tokenMeta == null) throw new PreCheckException(INVALID_TOKEN_ID);
+        if (tokenMeta == null) throw new PreCheckException(ResponseCodeEnum.INVALID_TOKEN_ID);
         if (tokenMeta.hasKycKey()) {
             context.requireKey(tokenMeta.kycKey());
+        } else {
+            throw new PreCheckException(TOKEN_HAS_NO_KYC_KEY);
         }
     }
 
     /**
      * This method is called during the handle workflow. It executes the actual transaction.
      *
-     * <p>Please note: the method signature is just a placeholder which is most likely going to
-     * change.
-     *
+     * @param txn           the {@link TransactionBody} of the active transaction
+     * @param tokenRelStore the {@link WritableTokenRelationStore} for the active transaction
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public void handle() {
-        throw new UnsupportedOperationException("Not implemented");
+    public void handle(@NonNull TransactionBody txn, @NonNull WritableTokenRelationStore tokenRelStore) {
+        requireNonNull(txn);
+        requireNonNull(tokenRelStore);
+
+        final var op = txn.tokenRevokeKycOrThrow();
+        pureChecks(op);
+
+        final var tokenId = op.tokenOrThrow().tokenNum();
+        final var accountId = op.accountOrElse(AccountID.DEFAULT).accountNumOrThrow();
+        final var tokenRel = tokenRelStore.getForModify(tokenId, accountId);
+
+        final var tokenRelBuilder = tokenRel.orElseThrow().copyBuilder();
+        tokenRelBuilder.kycGranted(false);
+        tokenRelStore.put(tokenRelBuilder.build());
+    }
+
+    private void pureChecks(TokenRevokeKycTransactionBody op) {
```

Comment:
These should be in `preHandle` and any checks that depend on `dynamicProperties` or state should be added to handle

Reply:
I'll move `pureChecks` to `preHandle`. Is anything missing from `handle`? I didn't see anything else in the current transition logic

Reply:
No handle seems fine for me

Reply:
This has the needed approvals and it's been open for awhile so I'll go ahead and merge, but I'll get these moved to `preHandle` in a PR today

Reply:
@Neeharika-Sompalli the move is in PR #6318